icaruscode/CRT/CRTPMTMatchingAna_module.cc

```diff
@@ -174,300 +312,391 @@ icarus::crt::CRTPMTMatchingAna::CRTPMTMatchingAna(fhicl::ParameterSet const& p)
+      flash_pos[1]=flash_pos[1]/ampsum;
+      flash_pos[2]=flash_pos[2]/ampsum;
+      t_m=t_m/nPMTsTriggering;
+      if( nPMTsTriggering <5 ) { continue; }
+      double gateDiff = (Long64_t)m_trigger_timestamp-(Long64_t)m_gate_start_timestamp;
+      bool inTime = flashInTime( firstTime, m_gate_type, gateDiff);
+      fRelGateTime=gateDiff+(tAbsflash-1500)*1e3;
+      fInTime_gate=false;
+      fInTime_beam=false;
+      if (m_gate_type==1 || m_gate_type==3){//BNB OffBeamBNB
+	if(fRelGateTime>-550 && fRelGateTime<2300) fInTime_gate=true;
+	if(fRelGateTime>-300 && fRelGateTime<1300) fInTime_beam=true;
+      }
+      if (m_gate_type==2 || m_gate_type==4){//NuMI OffBeamNuMI
+	if(fRelGateTime>-550 && fRelGateTime<10000) fInTime_gate=true;
+	if(fRelGateTime>-300 && fRelGateTime<9100) fInTime_beam=true;
+      }
+	inTime=fInTime_gate;
+    //  std::cout<<"Flash Time "<<tflash<<"  First Op Hit "<<firstTime<< "  " << inTime<<std::endl;
+    //  std::cout<<"Average Pos X "<<flash_pos[0]<<"  Y "<<flash_pos[1]<<"  Z "<<flash_pos[2]<<" nPMT "<<nPMTsTriggering<<" "<<ampsum<<std::endl;
+    //  std::cout<<"Flash X "<<flash->XCenter()<<"  "<<flash->YCenter()<<"  "<<flash->ZCenter()<<std::endl;
+      // Now get the CRT. Search the CRT Hits within -100 from the flash time
+      // NB currently the selection uses only top CRT. 
+      //  for the future a differentiation between Top and Side and some considerations based
+      //  on the proximity of the light are necessary 
+      //std::vector< art::Ptr<CRTHit> > selCRTHits;
+      icarus::crt::MatchedCRT CRTmatches = CRTHitmatched( firstTime, flash_pos, crtHitList, 0.1 );
+      int TopEn=0,TopEx=0, SideEn=0, SideEx=0;
+     
+      auto nCRTHits = CRTmatches.entering.size() + CRTmatches.exiting.size();
+   //   std::cout<<"Matched CRT "<<nCRTHits<<"  entering: "<<std::endl;
+  //    for (auto & entering : CRTmatches.entering){
+   //     std::cout<<"TOF "<<entering.tof<<"  distance  "<<entering.distance<<" velocity "<<(double)entering.distance/(double)entering.tof<<" Region "<<entering.CRTHit->plane<<std::endl;
+   //   }
+   //   std::cout<<"Exiting: "<<std::endl;
+   //   for (auto & exiting : CRTmatches.exiting){
+   //     std::cout<<"TOF "<<exiting.tof<<"  distance  "<<exiting.distance<<"  velocity "<<(double)exiting.distance/(double)exiting.tof<<" Region "<<exiting.CRTHit->plane<<std::endl;
+   //   }
+   //   std::cout<<std::endl;
+      bool matched=false;
+      double peflash=flash->TotalPE();
+      if (nCRTHits>0) matched=true;
+      if (matched==true){
+        eventType=CRTmatches.matchType;
+	for (auto & entering : CRTmatches.entering){
+          vector<double> CRTpos={entering.CRTHit->x_pos,entering.CRTHit->y_pos,entering.CRTHit->z_pos};
+          double CRTtime=entering.CRTHit->ts1_ns/1e3;
+          ULong64_t CRTAbsTime=entering.CRTHit->ts0_s;
+          int CRTRegion=entering.CRTHit->plane;
+          int CRTSys=0;
+          if (CRTRegion>=36) CRTSys=1;
```

Comment:
Could you comment what is special about CRTRegion greater or equal to 36?

Reply:
Yes. Very lazy way to determine if the Hit is Side CRT. I will update it soon with the bottom CRT.

Reply:
Alright, but could you put this comment in the code?

Reply:
Please disregard, I was looking at wrong version of the code!